sdk/storage/blobs/blob_client.h

```diff
@@ -0,0 +1,48 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#pragma once
+
+namespace azure { namespace storage { namespace blobs {
+
+class blob_client
+{
+public:
+    // connection string
+    blob_client(const std::string& connection_string, std::string container_name, std::string blob_name, blob_client_options options);
+
+    // shared key auth 
+    blob_client(azure::core::http::uri uri, azure::storage::common::storage_shared_key_credential credential, blob_client_options options);
+
+    // token auth
+    blob_client(azure::core::http::uri uri, azure::core::token_credential credential, blob_client_options options);
+
+    // anonymous/SAS/customized pipeline auth
+    blob_client(azure::core::http::uri uri, blob_client_options options);
+
+    bool is_snapshot() const;
+
+    azure::core::http::response<void> delete_(const delete_blob_options& options = delete_blob_options());
+
+    azure::core::http::response<bool> exists(const blob_exists_options& options = blob_exists_options());
+
+    azure::core::http::response<blob_properties> get_properties(const get_blob_properties_options& options = get_blob_properties_options());
+
+    azure::core::http::response<void> set_http_headers(const blob_http_headers& headers, set_blob_http_headers_options options = set_blob_http_headers_options());
+
+    azure::core::http::response<void> set_metadata(const std::map<std::string, std::string>& metadata, const set_blob_metadata_options = set_blob_metadata_options());
+
+    azure::core::http::response<blob_snapshot_info> create_snapshot(const create_blob_snapshot_options options = create_blob_snapshot_options());
+
+    azure::core::http::response<blob_download_info> download(const download_blob_options& options = download_blob_options());
+
+    // parallel download
+    blob_download_info download_to(std::ostream& destination, const download_blob_options& options = download_blob_options());
+    blob_download_info download_to(const std::string& path, const download_blob_options& options = download_blob_options());
+    blob_download_info download_to(char* buffer, uint64_t size, const download_blob_options& options = download_blob_options());
```

Comment:
`char*` or `uint8_t*`?

Reply:
I think `char*` is more common and can save users from one conversion.